Controllers/HandleAddInterventionErrors.swift

```diff
@@ -0,0 +1,127 @@
+      alert.message = "you_have_to_enter_phyto".localized
+    } else {
+      for selectedInput in selectedInputs {
+        if (selectedInput.value(forKey: "quantity") as? Double) == 0 {
+          alert.message = "you_have_to_enter_a_product_quantity".localized
+        }
+      }
+    }
+    if alert.message != "" {
+      alert.addAction(UIAlertAction(title: "ok".localized, style: .default, handler: nil))
+      present(alert, animated: true)
+      return false
+    }
+    return true
+  }
+
+  func fertilizationErrorHandler() -> Bool {
+    let alert = UIAlertController(title: "", message: "", preferredStyle: .alert)
+
+    if cropsView.selectedCrops.count == 0 {
+      alert.message = "you_have_to_select_a_crop".localized
+    } else if selectedInputs.count == 0 {
+      alert.message = "you_must_select_a_fertilizer".localized
+    } else {
+      for selectedInput in selectedInputs {
+        if (selectedInput.value(forKey: "quantity") as? Double) == 0 {
+          alert.message = "you_have_to_enter_a_product_quantity".localized
+        }
+      }
+    }
+    if alert.message != "" {
+      alert.addAction(UIAlertAction(title: "ok".localized, style: .default, handler: nil))
+      present(alert, animated: true)
+      return false
+    }
+    return true
+  }
+
+  func cropErrorHandler() -> Bool {
+    if cropsView.selectedCrops.count == 0 {
+      let alert = UIAlertController(title: "", message: "you_have_to_select_a_crop".localized, preferredStyle: .alert)
+
+      alert.addAction(UIAlertAction(title: "ok".localized, style: .default, handler: nil))
+      present(alert, animated: true)
+      return false
+    }
+    return true
+  }
+
+  func irrigationErrorHandler() -> Bool {
+    if cropsView.selectedCrops.count == 0 {
+      let alert = UIAlertController(title: "", message: "you_have_to_select_a_crop".localized, preferredStyle: .alert)
+
+      alert.addAction(UIAlertAction(title: "ok".localized, style: .default, handler: nil))
+      present(alert, animated: true)
+      return false
+    } else if irrigationValueTextField.text?.floatValue == 0 {
+      let alert = UIAlertController(title: "", message: "you_must_enter_a_water_volume".localized, preferredStyle: .alert)
+
+      alert.addAction(UIAlertAction(title: "ok".localized, style: .default, handler: nil))
+      present(alert, animated: true)
+      return false
+    }
+    return true
+  }
+
+  func checkErrorsInFunctionOfIntervention() -> Bool {
+    switch interventionType {
+    case Intervention.InterventionType.Care.rawValue.localized:
+      return cropErrorHandler()
+    case Intervention.InterventionType.CropProtection.rawValue.localized:
+      return cropProtectionErrorHandler()
+    case Intervention.InterventionType.Fertilization.rawValue.localized:
+      return fertilizationErrorHandler()
+    case Intervention.InterventionType.GroundWork.rawValue.localized:
+      return cropErrorHandler()
+    case Intervention.InterventionType.Harvest.rawValue.localized:
```

Comment:
You don't need this case because the execution code is the same as default's

Reply:
Can't hurt to let it here, makes it explicit how we treat that case instead of letting the reader wonder "did they miss it?"